features/step_definitions/index_page_steps.rb

```diff
@@ -0,0 +1,7 @@
+Given(/^I am on the index page$/) do
+  visit root_path
+end
+
+Given(/^a restaurant called "([^"]*)" exits$/) do |restaurant_name|
+  FactoryGirl.create(:restaurant, name: restaurant_name)
```

Comment:
add the FactoryGirl matchers and shorten the step def'w by omitting FactoryGirl when you call that in your definitions.